src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -831,13 +835,13 @@ private Tensor CastDataAndReturnAsTensor(T[] data)
                     return new Tensor((double[])(object)data, _dims, TF_DataType.TF_DOUBLE);
                 else if (typeof(T) == typeof(ReadOnlyMemory<char>))
                 {
-                    byte[][] bytes = new byte[_vBuffer.Length][];
-                    for (int i = 0; i < bytes.Length; i++)
+                    string[] strings = new string[_vBuffer.Length];
+                    for (int i = 0; i < strings.Length; i++)
                     {
-                        bytes[i] = Encoding.UTF8.GetBytes(((ReadOnlyMemory<char>)(object)data[i]).ToArray());
+                        strings[i] = new string(((ReadOnlyMemory<char>)(object)data[i]).ToArray());
```

Comment:
Just call `data[i].ToString()` instead. The advantage is that if the `ROM<char>` is backed by a full string, there is no need to allocate a new string object. 

[https://github.com/dotnet/runtime/blob/c5b6881e9b0e0826029185cd5a1569f80f6b9aba/src/libraries/System.Private.CoreLib/src/System/ReadOnlyMemory.cs#L148](https://github.com/dotnet/runtime/blob/c5b6881e9b0e0826029185cd5a1569f80f6b9aba/src/libraries/System.Private.CoreLib/src/System/ReadOnlyMemory.cs#L148)

The way you have it here, you will get an extra `char[]` object created, and an extra `string` created.